uobtheatre/bookings/models.py

```diff
@@ -521,36 +564,114 @@ def get_ticket_diff(
         )
 
     def pay(self, payment_method: "PaymentProvider") -> Optional["Transaction"]:
+        if self.is_reservation_expired:
+            raise CantBePaidForException(
+                message="This booking has expired. Please create a new booking"
+            )
+
+        # If this is a transfer and the booking it is transfered from is not paid
+        if self.transfered_from:
+            self.transfered_from.check_can_transfer_to(
+                self.performance
+            )  # pylint: disable=protected-access
+
+        return super().pay(payment_method)
+
+    def complete(self, payment: Transaction = None):
+        """
+        Complete the booking (after it has been paid for) and send the
+        confirmation email.
         """
-        Pay for booking using provided payment method.
+        super().complete()
 
-        Args:
-            payment_method (PaymentMethod): The payment method used to pay for
-                the booking
+        # If the booking is transfered from another
+        if self.transfered_from:
+            self.transfered_from.status = Booking.Status.CANCELLED
+            self.transfered_from.save()
 
-        Returns:
-            Payment: The payment created by the checkout (optional)
+        self.send_confirmation_email(payment)
```

Comment:
We will need a custom email for booking transfers

Reply:
You already appear to have done this at line 694 - I'm proud! 😉